control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -4762,7 +4768,18 @@ func createAWSDefaultSecurityGroup(ctx context.Context, ec2Client ec2iface.EC2AP
 	if len(vpcResult.Vpcs) == 0 {
 		return "", fmt.Errorf("vpc %s not found", vpcID)
 	}
-	vpcCIDR := awssdk.StringValue(vpcResult.Vpcs[0].CidrBlock)
+
+	// Validate the hostedcontrolplane defines a machine CIDR
+	if len(hcp.Spec.Networking.MachineNetwork) == 0 {
+		logger.Error(errors.New("hostedcontrolplane.spec.networking.machineNetwork length is 0"), "failed to extract machine CIDR while creating default security group")
```

Comment:
why are logging in addition to the error return here?

Reply:
Ah I was trying to conform to other areas in this function - I can align them to not duplicate logging + returning errors